### PR TITLE
Eng 7736/attempted login

### PR DESCRIPTION
**Pull request by cometman**

### CODE REVIEW

File: NeuroID/NeuroIDClass/Extensions/NIDUser.swift [hunk 1]
```diff
@@ -145,4 +145,21 @@ public extension NeuroID {
 
         return res
     }
+
+    /**
+        This should be called the moment a user trys to login. Returns true or false based on if a passed in UserID is valid.
+        @param {String} [attemptedRegisteredUserId] - an optional identifier for the login
+     */
+    static func attemptedLogin(_ attemptedRegisteredUserId:String? = nil) -> Bool{
+        if (attemptedRegisteredUserId != nil) {
```
Comment (kevinsites) [inline NeuroID/NeuroIDClass/Extensions/NIDUser.swift:154]: If the value is nil or fails validation we need to set the UID value like [trackjs does here LN 1551](https://github.com/Neuro-ID/trackjs/commit/d3eef8c3894b2e831f3ebf1e48fba168380ae363)

**Review by kevinsites — CHANGES_REQUESTED** [review]

File: NeuroID/NeuroIDClass/Extensions/NIDUser.swift [hunk 2]
```diff
@@ -145,4 +145,26 @@ public extension NeuroID {
...
+            if (!NeuroID.validateUserID(attemptedRegisteredUserId!)) {
+                isValidID = false
+            } else {
+                isValidID = true
+            }
+        }
+            
+        let hashedUserId = isValidID
+        ? attemptedRegisteredUserId?.hashValue() : "scrubbed-id-failed-validation"
+
```
Comment (kevinsites) [inline NeuroID/NeuroIDClass/Extensions/NIDUser.swift:165]: Consider instead
``` 
let hashedUserId:String
if (!NeuroID.validateUserID(attemptedRegisteredUserId ?? "")) {
    hashedUserId = attemptedRegisteredUserId!.hashValue()
} else {
    hashedUserId = "scrubbed-id-failed-validation"
}
```
 so we don't have to do a null check and validation check but rather let validation do all of it. Also removes the need for the `isValidID` variable

**Review by kevinsites — CHANGES_REQUESTED** [review]

File: NeuroID/NeuroIDClass/Extensions/NIDUser.swift (hunk 2 above)
Comment (kevinsites) [reply NeuroID/NeuroIDClass/Extensions/NIDUser.swift:165]: where would it be duplicating?

Comment (cometman) [reply NeuroID/NeuroIDClass/Extensions/NIDUser.swift:165]: Saving to the datastore and the return of the validation if you put it in that block. If you had another idea mind showing me ?

**Review by kevinsites — APPROVED** [review]

**Review by kishisaka-nid — APPROVED** [review]
LGTM! 🚢

**Pull request merged**